File: 0004.java
public class _0004 {
    public Object fn0() {
        int max = 0;
        for (int i = 100; i < 1000; i++) {
            for (int j = 100; j < 1000; j++) {
                String s = Integer.toString(i * j);
                for (int a = 0, b = s.length()-1; a < b; a++, b--) {
                    if (s.charAt(a) != s.charAt(b)) break;
                    if (a == b || a+1 >= b-1) {
                        if (i * j > max) {
                            max = i * j;
                        }
                    }
                }
            }
        }
        return max;
    }

    public String[] solutions = new String[]{ "fn0" };
}

File: 0004.py
def ispalindromic(n):
    r = str(n)
    if len(r) % 2:
        return False
    lim = len(r) / 2
    for i in range(lim):
        if r[i] != r[-(i+1)]:
            return False
    return True

def fn0():
    largestp = 0
    for i in xrange(100, 1000):
        for j in xrange(100, 1000):
            r = i * j
            if ispalindromic(r) and r > largestp:
                #print r
                largestp = r
    return largestp

solutions = [fn0]
